include/swift/AST/ASTContext.h

```diff
@@ -368,6 +368,12 @@ class ASTContext final {
   /// i.e. true if the entry is [key: alias_name, value: (real_name, true)].
   mutable llvm::DenseMap<Identifier, std::pair<Identifier, bool>> ModuleAliasMap;
 
+  /// The maximum arity of `_StringProcessing.Tuple{n}`.
+  static constexpr unsigned StringProcessingTupleDeclMaxArity = 8;
+  /// Cached `_StringProcessing.Tuple{n}` declarations.
+  mutable SmallVector<StructDecl *, StringProcessingTupleDeclMaxArity - 2>
+      StringProcessingTupleDecls;
```

Comment:
Could this be a fixed-size C array?

Reply:
I considered that initially but I preferred not having to eagerly initialize it until we need it. Since we never go beyond `StringProcessingTupleDeclMaxArity - 2`, there won't be any heap allocations with a small vector.